app/policies/proposal_policy.rb

```diff
@@ -35,4 +40,30 @@ def can_edit!
   end
 
   alias_method :can_update!, :can_edit!
+
+  def can_show!
+    check(author? || approver? || observer?,
```

Comment:
Wondering if we should simplify this to `@proposal.users.include?(user)`, so that it (eventually) will be one SQL call instead of three.